packages/components/src/components/SquigglePlayground/index.tsx

```diff
@@ -155,119 +154,151 @@ export const SquigglePlayground: React.FC<PlaygroundProps> = (props) => {
 
   const editorRef = useRef<CodeEditorHandle>(null);
 
-  const firstTab = showEditor ? (
-    <div className="border border-slate-200" data-testid="squiggle-editor">
-      <CodeEditor
-        ref={editorRef}
-        value={code}
-        errors={errors}
-        project={resultAndBindings.project}
-        showGutter={true}
-        height={height}
-        onChange={setCode}
-        onSubmit={runnerState.run}
-      />
-    </div>
-  ) : (
-    squiggleChart
+  const standardHeightStyle = (height) => ({ height, overflow: "auto" });
+  const { ref: leftSideHeader, height: leftSideHeaderHeight } = useHeight();
+  const { ref: rightSideHeader, height: rightSideHeaderHeight } = useHeight();
```

Comment:
Good point.

Any tips around how to render in SSR? Are there other hooks that do work, or should we not rely on this kind of height-adjustment in general?

Having fixed header heights will be annoying, but I could understand the benefit in this case.

Reply:
Anything with `useEffect` is problematic. In general, `useEffect` is discouraged by the new React docs: https://react.dev/learn/you-might-not-need-an-effect.

Also, any layout adjustments made on client are bad because of https://web.dev/cls/.